ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx

```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
   padding: var(--margins-sm) var(--margins-2xl) var(--margins-lg);
 `;
 
+type MessagePropsDetails = {
+  errors: Array<Error>;
+  attachments: Array<PropsForAttachment>;
+};
+
+async function getPropsForMessageDetail(
+  messageId: string | undefined,
+  attachments: Array<PropsForAttachment>
+): Promise<MessagePropsDetails | null> {
+  if (!messageId) {
+    return null;
+  }
+  const found = await Data.getMessageById(messageId);
+  const attachmentsWithMediaDetails: Array<PropsForAttachment> = [];
+  if (found) {
+    // process attachments so we have the fileSize, url and screenshots
+    for (let i = 0; i < attachments.length; i++) {
+      const props = found.getPropsForAttachment(attachments[i]);
+      if (
+        props?.contentType &&
+        GoogleChrome.isVideoTypeSupported(props?.contentType) &&
+        !props.duration &&
+        props.url
+      ) {
+        // eslint-disable-next-line no-await-in-loop
+        const duration = await getVideoDuration({
+          objectUrl: props.url,
+          contentType: props.contentType,
+        });
+        attachmentsWithMediaDetails.push({
+          ...props,
+          duration,
+        });
+      } else if (props?.contentType && isAudio(props.contentType) && !props.duration && props.url) {
+        // eslint-disable-next-line no-await-in-loop
+        const duration = await getAudioDuration({
+          objectUrl: props.url,
+          contentType: props.contentType,
+        });
+
+        attachmentsWithMediaDetails.push({
+          ...props,
+          duration,
+        });
+      } else if (props) {
+        attachmentsWithMediaDetails.push(props);
+      }
+    }
+
+    // This will make the error message for outgoing key errors a bit nicer
+    const errors = (found.get('errors') || []).map((error: any) => {
+      return error;
+    });
+
+    const toRet: MessagePropsDetails = {
+      errors,
+      attachments: attachmentsWithMediaDetails,
+    };
+
+    return toRet;
+  }
+  return null;
+}
+
+function useMessageDetailProps(messageId: string | undefined): MessagePropsDetails | null {
+  const [details, setDetails] = useState<MessagePropsDetails | null>(null);
+
+  const fromState = useMessageAttachments(messageId);
+
+  // this is not ideal, but also doesn't seem to create any performance issue at the moment.
+  // TODO: ideally, we'd want to save the attachment duration anytime we save one to the disk (incoming/outgoing), and just retrieve it from the redux state here.
+  useEffect(() => {
```

Comment:
Correct me if I am wrong but is this not a case for `useMount`?

Reply:
I don't think so.
I think useMount wouldn't reload the data if we keep the right panel open but switch between two messages right

Reply:
True. Resolved